.github/workflows/ci.yml

```
@@ -0,0 +1,56 @@
+name: GitHub CI
+
+on:
+  push:
+  pull_request:
+    paths-ignore:
+      - 'README.md'
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [Windows-latest, Ubuntu-latest, macOS-latest]
+        configuration: [Debug, Release]
+        include:
+          - os: windows-latest
+            cmake-parameters: -G "Visual Studio 16 2019" -A x64
+            folder: win64
+          - os: ubuntu-latest
+            cmake-parameters: -G "Unix Makefiles"
+            folder: linux64
+          - os: macos-latest
+            cmake-parameters: -G "Xcode"
+            folder: macos64
+    steps:
+      - uses: actions/checkout@v2
+      - name: Generate CMake Files
+        run: |
+          mkdir build && cd build
+          cmake .. ${{ matrix.cmake-parameters }}
+      - name: Build
+        working-directory: build
+        run: cmake --build . --config ${{ matrix.configuration }}
+      - name: Copy Files
+        if: matrix.configuration == 'Release'
+        working-directory: build
+        run: cmake --install . --config ${{ matrix.configuration }} --prefix ../upload/${{ matrix.folder }}
```

If you change it to this you won't need the folder include:
```suggestion
        run: cmake --install . --config ${{ matrix.configuration }} --prefix ../upload/${{ runner.os }}64
```

---

What about other architectures? I would like to have both platforms and architecture combination remain possible than hardcode specific architecture. Which means if I do your way will have to revert back and make sure nothing else is broken.

I may will add ARM architecture in the future if there are requests for it.

[Also runner.os from the doc says it would be Windows, Linux, macOS](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#runner-context) So, there's no shorter name at runtime.

---

You could add `arch`, but that might be too convoluted.

---

Hmm... different architectures will need go into a different job section in order for it to work which require duplicate steps. Or by using different os name which I'm not really interest in.

---

Resolved

Instead of `64`, it will be `_x64` prefix. To separate platform and architecture readability.